halo/attest/voter/voter.go

```diff
@@ -191,6 +191,7 @@ func (v *Voter) runOnce(ctx context.Context, chainVer xchain.ChainVersion) error
 		Offset:    fromOffset,
 	}
 
+	cursors := make(map[xchain.StreamID]uint64, len(v.network.Chains))
```

Comment:
As far as I can tell, we shouldn't need a lock over this, because only one process can ever access this map. Happy to throw one on anyway if we think it's advisable.

Reply:
Correct, lock is not required.

`len(v.network.Chains)` is incorrect, since we have multiple shards per destination chain per source chain. Best avoid setting length in this case.